newrelic/hooks/external_botocore.py

```diff
@@ -40,43 +56,232 @@ def extractor_string(*args, **kwargs):
     return extractor_list
 
 
+def check_rate_limit_header(response_headers, header_name, is_int):
+    if not response_headers:
+        return ""
+
+    if header_name in response_headers:
+        header_value = response_headers.get(header_name)
+        if is_int:
+            header_value = int(header_value)
+        return header_value
+    else:
+        return ""
+
+
+def create_chat_completion_message_event(
+    transaction,
+    app_name,
+    message_list,
+    chat_completion_id,
+    span_id,
+    trace_id,
+    request_model,
+    request_id,
+    conversation_id,
+    response_id="",
+):
+    if not transaction:
+        return
+
+    for index, message in enumerate(message_list):
+        if response_id:
+            id_ = "%s-%d" % (response_id, index)  # Response ID was set, append message index to it.
+        else:
+            id_ = str(uuid.uuid4())  # No response IDs, use random UUID
+
+        chat_completion_message_dict = {
+            "id": id_,
+            "appName": app_name,
+            "conversation_id": conversation_id,
+            "request_id": request_id,
+            "span_id": span_id,
+            "trace_id": trace_id,
+            "transaction_id": transaction._transaction_id,
+            "content": message.get("content", "")[:MAX_LOG_MESSAGE_LENGTH],
```

Comment:
This happens inside the attribute validation now.
```suggestion
            "content": message.get("content", ""),
```